submissions/MarharytaBoichenko/oop/index.js

```diff
@@ -0,0 +1,78 @@
+class Inhabitant {
+  constructor(name, species, friends, saying) {
+    this.name = name;
+    this.species = species;
+    this.friends = friends;
+    this.saying = saying;
+    this.properties = ["name", "species", "friends", "saying"];
+  }
+  toPrint() {
+    return this.properties
+      .map((prop) => {
+        const neededProp = this[prop] === undefined ? `no ${prop}` : this[prop];
+        return `my ${prop}: ` + neededProp;
+      })
+      .join(", ");
+  }
+}
+
+class Person extends Inhabitant {
+  constructor(name, gender, friends, saying) {
+    super(name, "human", friends, saying);
+    this.gender = gender;
+    this.hands = 2;
+    this.legs = 2;
+    this.properties = [...this.properties, "gender", "hands", "legs"];
+  }
+}
+
+class Man extends Person {
+  constructor(name, friends, saying) {
+    super(name, "man", friends, saying);
+  }
+}
+
+class Woman extends Person {
+  constructor(name, friends, saying) {
+    super(name, "woman", friends, saying);
+  }
+}
+class Animal extends Inhabitant {
+  constructor(name, species, legs, gender, friends, saying) {
+    super(name, species, friends, saying);
+    this.legs = legs;
+    this.gender = gender;
```

Comment:
What makes `legs` and `gender` different from `name` in terms of handling and generalization of shared properties?

Reply:
I changed  gender, but not  legs,  I think  it differs  because here class Animal extends Inhabitant   in parents  class Inhabitant  there  no  legs.

Reply:
Why `legs` do not deserve the same as `name`?

Reply:
I think  my  mistake is that  I write legs for  Animal  but thought about legs  as  differnt properties  ,  not as legs  which  Person has  ). As  I  understand  if  it legs  and this  property  is  in Animal  and in  Person  too so we  need handle it  as  name and  gender  and legs  will go  from  parent  class.  And  if I decide  that legs  in class  Animals  is  diffferent property  - paws so in Animal it will be this.paw =  paw . Is  it  right?

Reply:
Exactly

Reply:
thanks, done